### PR TITLE
730: Reset Qt response and Qt related tests

### PR DESCRIPTION
**Pull request by tomlovesgithub**
## What's included?

This PR includes changes which support candidates ability to retake a QT if they have started and then been reset through the admin portal.
I have also included some unit tests for the tab functionality and created a working pattern to mock dates.
I have removed the unused `countdown` component and changed the `countdown2` to be the only QT timer, i have also added an initial test for the countdown.

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
 To test you must have an open QT with access to a candidate account eligible to take the qualifying test.
 - Complete qualifying test. 
 - Observe the test shows in past tab, not in open and cannot be restarted.
 - Use accompanying #1226 admin ticket to reset candidate response.
 - Observe the test showing in open tests; can be started.
 - Resubmit test.

## Risk - how likely is this to impact other areas?
🟠 Medium risk - this does change code that is shared with other areas

## Additional context
image

Video: [here](https://drive.google.com/drive/u/1/folders/1-kkQ1_ArvpIiHIid7NBPDzwnPoss0TPR)

https://user-images.githubusercontent.com/44227249/126500643-c7ca230c-4f1b-4c95-8ba4-2ff4f13257d6.mov

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

File: src/helpers/date.js [hunk 1]
```diff
@@ -87,26 +87,22 @@ const helperTimeLeft = (obj) => {
...
-        return duration * minute;
-      }
-      const endTime = new Date(startTime.getTime() + duration * minute);
-      if (endTime < Date.now()) {
-        return 0;
-      }
-      return (endTime - Date.now());
+    const endTime = new Date(startTime.getTime() + duration * minute);
+    if (endTime < Date.now()) {
+      return 0;
```
Comment (mbrookeswebdev) [inline src/helpers/date.js:103]: Why are you returning 0 and not false?

**Review by HalcyonJAC — COMMENTED** [review]
Hi @tomlovesgithub

I was looking into this one, the code looks good but I can't find a 'Reset' button as such. Can you send me some instructions it please?

PS: I have created a QT int the Admin portal and started it in the Apply application but just can't seem to find a Reset button anywhere...

**Review by mbrookeswebdev — APPROVED** [review]

**Comment by KateMJAC** [comment]
Ticket to be discussed at Friday stand-up on 2 July. Advised by Tom that we have the option to proceed with this work or roll what has been done back. 

We need consensus as to the value of the work and whether its felt we should proceed.

**Comment by HalcyonJAC** [comment]
Hi @tomlovesgithub please could you provide some info on how to test? Thanks!

**Review by HalcyonJAC — APPROVED** [review]

**Review by warrensearle — APPROVED** [review]
Great work @tomlovesgithub and 👍 for tidying up the existing code and including some tests 😄

**Review by joy-ade — COMMENTED** [review]
Out to UTG 22/07/2021

**Review by joy-ade — APPROVED** [review]
Approved by Product team

**Pull request merged**